contentcuration/contentcuration/static/js/edit_channel/preview/views.js

```diff
@@ -24,43 +24,88 @@ var PreviewModalView = BaseViews.BaseModalView.extend({
 var PreviewView = BaseViews.BaseView.extend({
     tabs_template: require("./hbtemplates/preview_templates/tabs.handlebars"),
     template: require("./hbtemplates/preview_dialog.handlebars"),
-    current_preview:null,
     initialize: function(options) {
-        _.bindAll(this, 'set_preview','toggle_fullscreen', 'load_preview', 'exit_fullscreen');
-        this.presets = new Models.FormatPresetCollection();
-        this.questions = new Models.AssessmentItemCollection();
+        _.bindAll(this, 'select_preview','toggle_fullscreen', 'load_preview', 'exit_fullscreen', 'render_preview');
+        this.current_preview = null;
         this.render();
     },
     events: {
-        'click .preview_btn_tab' : 'set_preview',
+        'click .preview_btn_tab' : 'select_preview',
         'click .view_fullscreen': 'toggle_fullscreen'
     },
     render: function() {
+        this.load_preview();
         this.$el.html(this.template({
-            node: (this.model)? this.model.toJSON() : null,
-            file: this.current_preview,
-            selected_preset: (this.current_preview) ?  window.formatpresets.get(this.current_preview.preset).toJSON() : null,
+            file: this.current_preview
         }));
+        this.load_preset_dropdown();
+        this.render_preview();
+    },
+    render_preview:function(){
+        if(this.current_preview){
+            if (this.current_preview.preset && this.current_preview.preset.readable_name){
+                this.$("#preview_format_switch").text(this.current_preview.preset.readable_name);
+            }else{
+                this.$("#preview_format_switch").text("Q: " + this.current_preview.question);
```

Comment:
I'm not sure if it's relevant here, but I usually prefer class selectors to id selectors when possible. This ensures that even if a template is rendered multiple times on the same page, there ids are never duplicated across DOM nodes.